integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -81,6 +82,115 @@ fn check_state_shard_uid_mapping_after_resharding(client: &Client, parent_shard_
     }
 }
 
+struct TestReshardingParameters {
+    chunk_ranges_to_drop: HashMap<ShardUId, std::ops::Range<i64>>,
+    accounts: Vec<AccountId>,
+    clients: Vec<AccountId>,
+    block_and_chunk_producers: Vec<AccountId>,
+    initial_balance: u128,
+    epoch_length: BlockHeightDelta,
+    /// Custom behavior executed at every iteration of test loop.
+    loop_action: Option<Box<dyn Fn(&mut TestLoopData, TestLoopDataHandle<ClientActorInner>)>>,
```

Comment:
As there is only one such action, I'd go with `should_fork_before_resharding_block`. If there is likelihood that it will always have one action, we'll save one level of unnecessary abstraction. And extending it to `loop_action` again if needed is an easy refactor. It's harder to go back, especially for an unfamiliar engineer.

Reply:
Currently there's already two different actions both produced by `fork_before_resharding_block()`. I would need two bools to express the same functionality or an enum. For this reason I'm slightly in favor of keeping `loop_action`.

Reply:
We can store this as a bool parameter (`double_signing: bool`). But well, that's quite a minor thing